client/state/ui/oauth2-clients/selectors.js

```diff
@@ -10,13 +10,13 @@ import { get } from 'lodash';
  * @return {Object}          OAuth2 client data
  */
 export const getOAuth2ClientData = ( state ) => {
-	const currentClientId = get( state, 'login.oauth2.currentClientId', null );
+	const currentClientId = get( state, 'ui.oauth2Clients.currentClientId', null );
 
 	if ( ! currentClientId ) {
 		return null;
 	}
 
-	return get( state, 'login.oauth2.clients', {} )[ currentClientId ];
+	return get( state, 'ui.oauth2Clients.clients', {} )[ currentClientId ];
```

Comment:
My suggestion would be to have `oauth2Clients` as a list at the root of the state tree. And `ui` would have everything related to the current client id, a bit like `users` and `currentUserId` at the moment.
With this organisation,
- `getOAuth2ClientData` would live in `state/ui/oauth2-clients/selectors.js`
- `showOAuth2Layout` would live in `state/ui/oauth2-clients/selectors.js` (we could probably rename it btw as it sounds like an action)
- `currentClientId` would live in `state/ui/oauth2-clients/reducer.js` like it already does
- `oauth2Clients` would live in `state/oauth2-clients/reducer.js`